sdk/notificationhubs/aznotificationhubs/client.go

```diff
@@ -0,0 +1,347 @@
+	requestUri := fmt.Sprintf("%v%v/installations/%v?api-version=%v", n.endpointUrl, n.hubName, installationId, AZNHApiVersion)
+
+	client := &http.Client{Timeout: time.Second * 15}
+	req, err := http.NewRequest(http.MethodGet, requestUri, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	n.addRequestHeaders(req)
+
+	res, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode != 200 {
+		return nil, fmt.Errorf("invalid response from Azure Notification Hubs: %v", res.StatusCode)
+	}
+
+	defer res.Body.Close()
+
+	var installation Installation
+	decoder := json.NewDecoder(res.Body)
+	err = decoder.Decode(&installation)
+	if err != nil {
+		return nil, err
+	}
+
+	return &installation, nil
+}
+
+// Creates or updates an installation.
+func (n *NotificationHubClient) CreateOrUpdateInstallation(installation *Installation) (*NotificationResponse, error) {
+	requestUri := fmt.Sprintf("%v%v/installations/%v?api-version=%v", n.endpointUrl, n.hubName, installation.InstallationId, AZNHApiVersion)
+
+	installationJSON, err := json.Marshal(installation)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{Timeout: time.Second * 15}
+	req, err := http.NewRequest(http.MethodPut, requestUri, bytes.NewBuffer(installationJSON))
+	if err != nil {
+		return nil, err
+	}
+
+	n.addRequestHeaders(req)
+	req.Header.Add("Content-Type", "application/json")
+
+	res, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode != 200 {
+		return nil, fmt.Errorf("invalid response from Azure Notification Hubs: %v", res.StatusCode)
+	}
+
+	return createNotificationResponse(res)
+}
+
+// Updates an installation with the specified patches.
+func (n *NotificationHubClient) UpdateInstallation(installationId string, patches []InstallationPatch) (*NotificationResponse, error) {
+	requestUri := fmt.Sprintf("%v%v/installations/%v?api-version=%v", n.endpointUrl, n.hubName, installationId, AZNHApiVersion)
+
+	patchesJSON, err := json.Marshal(patches)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{Timeout: time.Second * 15}
+	req, err := http.NewRequest(http.MethodPatch, requestUri, bytes.NewBuffer(patchesJSON))
+	if err != nil {
+		return nil, err
+	}
+
+	n.addRequestHeaders(req)
+	req.Header.Add("Content-Type", "application/json-patch+json")
+
+	res, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode != 200 {
+		return nil, fmt.Errorf("invalid response from Azure Notification Hubs: %v", res.StatusCode)
+	}
+
+	return createNotificationResponse(res)
+}
+
+// Deletes an installation from Azure Notification Hubs.
+func (n *NotificationHubClient) DeleteInstallation(installationId string) (*NotificationResponse, error) {
+	requestUri := fmt.Sprintf("%v%v/installations/%v?api-version=%v", n.endpointUrl, n.hubName, installationId, AZNHApiVersion)
+
+	client := &http.Client{Timeout: time.Second * 15}
```

Comment:
We will need to move to `azcore` to use its client pipeline before release.

Reply:
Done